### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by MartinYSpasov**

### CODE REVIEW

File: stream-transformation-test/stream-transformation-it/src/test/java/uk/gov/justice/framework/tools/replay/StreamTransformationIT.java [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+    @Before
+    public void setUpDB() throws Exception {
+        dataSource = initEventStoreDb();
+        EVENT_LOG_JDBC_REPOSITORY = new TestEventLogJdbcRepository(dataSource);
+        EVENT_STREAM_JDBC_REPOSITORY = new TestEventStreamJdbcRepository(dataSource);
+    }
+
+    @Test
+    public void runReplayTool() throws Exception {
```
Comment (DesMarshall01) [inline stream-transformation-test/stream-transformation-it/src/test/java/uk/gov/justice/framework/tools/replay/StreamTransformationIT.java]: Would be better named as runTransformationTool()

File: event-tool/pom.xml [hunk 2]
```diff
@@ -50,6 +50,11 @@
             <groupId>org.wildfly.swarm</groupId>
             <artifactId>datasources</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.h2database</groupId>
+            <artifactId>h2</artifactId>
+            <version>${com.h2database.version}</version>
```
Comment (matt-rich) [inline event-tool/pom.xml]: Do you need the version here?

File: pom.xml [hunk 3]
```diff
@@ -18,9 +18,12 @@
         <framework.version>3.0.0-SNAPSHOT</framework.version>
         <common-bom.version>1.21.0</common-bom.version>
         <utilities.version>1.10.0</utilities.version>
-        <test-utils.version>1.13.1</test-utils.version>
+        <test-utils.version>1.15.0</test-utils.version>
         <wildfly.swarm.version>2017.11.0</wildfly.swarm.version>
         <version.swarm.fraction-plugin>77</version.swarm.fraction-plugin>
+        <test.utils.core.version>1.15.0</test.utils.core.version>
+        <uk.gov.justice.utils.version>1.5.0</uk.gov.justice.utils.version>
+        <com.h2database.version>1.4.187</com.h2database.version>
```
Comment (matt-rich) [inline pom.xml]: Is this not set in the maven parent pom somewhere? or common bom.

File: stream-transformation-test/sample-transformations/pom.xml [hunk 4]
```diff
@@ -22,6 +24,14 @@
         <dependency>
             <groupId>uk.gov.justice.utils</groupId>
             <artifactId>test-utils-core</artifactId>
+            <version>${test.utils.core.version}</version>
```
Comment (matt-rich) [inline stream-transformation-test/sample-transformations/pom.xml]: we can add these to the managed dependencies list in the repos top level pom, then we can exclude the version.

File: stream-transformation-test/stream-transformation-it/pom.xml [hunk 5]
```diff
@@ -12,4 +12,164 @@
 
     <artifactId>stream-transformation-it</artifactId>
 
+    <dependencies>
+
+        <dependency>
+            <groupId>com.h2database</groupId>
+            <artifactId>h2</artifactId>
+            <scope>test</scope>
+            <version>${com.h2database.version}</version>
```
Comment (matt-rich) [inline stream-transformation-test/stream-transformation-it/pom.xml]: version required?

File: stream-transformation-test/sample-transformations/pom.xml [hunk 6]
```diff
@@ -22,6 +24,14 @@
...
+            <exclusions>
```
Comment (matt-rich) [inline stream-transformation-test/sample-transformations/pom.xml]: Why do we need the exclusion? Is the test utils using a different version? we should probably try and make it consistent across all of them by updating the common bom perhaps.

File: stream-transformation-test/stream-transformation-it/pom.xml [hunk 7]
```diff
@@ -12,4 +12,164 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>uk.gov.justice.utils</groupId>
```
Comment (matt-rich) [inline stream-transformation-test/stream-transformation-it/pom.xml]: if we set this in managed dependencies then we shouldn't need to do exclusions each time (I think)

File: stream-transformation-test/stream-transformation-it/pom.xml [hunk 8]
```diff
@@ -12,4 +12,164 @@
...
+                    <groupId>com.h2database</groupId>
+                    <artifactId>h2</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+
+        <dependency>
+            <groupId>uk.gov.justice.services</groupId>
+            <artifactId>test-utils-core</artifactId>
+            <version>${framework.version}</version>
```
Comment (matt-rich) [inline stream-transformation-test/stream-transformation-it/pom.xml]: set in managed dependencies, remove version here.

File: stream-transformation-test/stream-transformation-it/pom.xml [hunk 9]
```diff
@@ -12,4 +12,164 @@
...
+        <dependency>
+            <groupId>org.hamcrest</groupId>
+            <artifactId>hamcrest-library</artifactId>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>uk.gov.justice.services</groupId>
+            <artifactId>event-repository-liquibase</artifactId>
+            <version>${framework.version}</version>
```
Comment (matt-rich) [inline stream-transformation-test/stream-transformation-it/pom.xml]: managed dependencies, remove version

File: stream-transformation-test/stream-transformation-it/pom.xml [hunk 10]
```diff
@@ -12,4 +12,164 @@
...
+            <artifactId>event-repository-jdbc</artifactId>
+            <version>${framework.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>uk.gov.justice.utils</groupId>
+            <artifactId>test-utils-logging-simple</artifactId>
+            <type>pom</type>
+            <scope>test</scope>
+            <version>${uk.gov.justice.utils.version}</version>
```
Comment (matt-rich) [inline stream-transformation-test/stream-transformation-it/pom.xml]: managed dependencies.. and above. We shouldn't need to be specifying versions.

File: stream-transformation-test/stream-transformation-it/src/test/java/uk/gov/justice/framework/tools/replay/StreamTransformationIT.java (hunk 1 above)
Comment (matt-rich) [inline stream-transformation-test/stream-transformation-it/src/test/java/uk/gov/justice/framework/tools/replay/StreamTransformationIT.java]: runEventTransformationTool

We should use normal test naming really, shouldTransformEventInEventStore()

File: stream-transformation-test/stream-transformation-it/src/test/java/uk/gov/justice/framework/tools/replay/StreamTransformationIT.java [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+        insertEventLogData();
+
+        runCommand(createCommandToExecuteStreamReplay());
+
+        assertTrue(eventStoreTransformedEventPresent());
+        assertTrue(originalEventStreamIsActive());
+        assertFalse(clonedStreamIsInactive());
```
Comment (matt-rich) [inline stream-transformation-test/stream-transformation-it/src/test/java/uk/gov/justice/framework/tools/replay/StreamTransformationIT.java]: Think we need to change wording, assert False on Inactive suggests that it is active.
The cloned should be assertTrue(clonedIsInactive) or assertFalse(clonedIsActive)

File: stream-transformation-test/stream-transformation-it/src/test/java/uk/gov/justice/framework/tools/replay/StreamTransformationIT.java [hunk 12]
```diff
@@ -0,0 +1,210 @@
...
+
+        final boolean isActive = EVENT_STREAM_JDBC_REPOSITORY.findAll()
+                .filter(eventStream -> eventStream.getStreamId().equals(clonedStreamId))
+                .findFirst().get().isActive();
+
+        return isActive;
+    }
+
+    private boolean originalEventStreamIsActive() {
+        final Stream<EventStream> eventStream = EVENT_STREAM_JDBC_REPOSITORY.findAll();
```
Comment (matt-rich) [inline stream-transformation-test/stream-transformation-it/src/test/java/uk/gov/justice/framework/tools/replay/StreamTransformationIT.java]: is this the same method as above? essentially isActive(streamId) just written in two ways. Maybe we can refactor this so it's a reusable method.

File: stream-transformation-test/stream-transformation-it/src/test/java/uk/gov/justice/framework/tools/replay/StreamTransformationIT.java [hunk 13]
```diff
@@ -0,0 +1,210 @@
...
+
+        final UUID streamId = metadata.streamId().get();
+        final String name = metadata.name();
+        final String payload = jsonEnvelope.payloadAsJsonObject().toString();
+        final ZonedDateTime createdAt = metadata.createdAt().get();
+
+        return new Event(id, streamId, sequenceId, name, metadata.asJsonObject().toString(), payload, createdAt);
+    }
+
+    private String createCommandToExecuteStreamReplay() throws IOException {
```
Comment (matt-rich) [inline stream-transformation-test/stream-transformation-it/src/test/java/uk/gov/justice/framework/tools/replay/StreamTransformationIT.java]: streamReplay still in name.

File: stream-transformation-test/stream-transformation-it/src/test/java/uk/gov/justice/framework/tools/replay/StreamTransformationIT.java [hunk 14]
```diff
@@ -0,0 +1,210 @@
...
+            if (!dropped) {
+                liquibase.dropAll();
+                dropped = true;
+            }
+            liquibase.update("");
+        }
+        return dataSource;
+    }
+
+    public boolean eventStoreTransformedEventPresent() throws SQLException {
```
Comment (matt-rich) [inline stream-transformation-test/stream-transformation-it/src/test/java/uk/gov/justice/framework/tools/replay/StreamTransformationIT.java]: need to be public?

File: stream-transformation-test/stream-transformation-it/src/test/java/uk/gov/justice/framework/tools/replay/StreamTransformationIT.java [hunk 15]
```diff
@@ -0,0 +1,210 @@
...
+        final Stream<Event> eventLogs = EVENT_LOG_JDBC_REPOSITORY.findAll();
+        final Optional<Event> event = eventLogs.filter(item -> item.getStreamId().equals(STREAM_ID)).findFirst();
+
```
Comment (matt-rich) [inline stream-transformation-test/stream-transformation-it/src/test/java/uk/gov/justice/framework/tools/replay/StreamTransformationIT.java]: I think we need to check the event itself has been transformed.

File: stream-transformation-test/stream-transformation-it/src/test/java/uk/gov/justice/framework/tools/replay/StreamTransformationIT.java [hunk 16]
```diff
@@ -0,0 +1,210 @@
...
+                new BufferedReader(new InputStreamReader(exec.getInputStream()));
+
+        String line = "";
+        while ((line = reader.readLine()) != null) {
+            System.out.println(line);
+        }
+    }
+
+    private void insertEventLogData() throws SQLException, InvalidSequenceIdException {
+        EVENT_LOG_JDBC_REPOSITORY.insert(eventLogFrom("people.events.person-gender-updated", 1L));
```
Comment (matt-rich) [inline stream-transformation-test/stream-transformation-it/src/test/java/uk/gov/justice/framework/tools/replay/StreamTransformationIT.java]: we should change this to not refer to people. (yes I know I put it in there in the first place :))

I've made a change in the sampleTransformations locally, you'd need to keep them in sync.

File: stream-transformation-test/stream-transformation-it/src/test/java/uk/gov/justice/framework/tools/replay/StreamTransformationIT.java [hunk 17]
```diff
@@ -0,0 +1,210 @@
...
+        EVENT_STREAM_JDBC_REPOSITORY.insert(STREAM_ID ,true);
```
Comment (matt-rich) [inline stream-transformation-test/stream-transformation-it/src/test/java/uk/gov/justice/framework/tools/replay/StreamTransformationIT.java]: i think there is a method with a single parameter, by default it sets as active.

File: stream-transformation-test/stream-transformation-it/src/test/java/uk/gov/justice/framework/tools/replay/TestEventLogJdbcRepository.java [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+import uk.gov.justice.services.eventsourcing.repository.jdbc.AnsiSQLEventLogInsertionStrategy;
+import uk.gov.justice.services.eventsourcing.repository.jdbc.event.EventJdbcRepository;
+import uk.gov.justice.services.jdbc.persistence.JdbcRepositoryHelper;
+
+import javax.sql.DataSource;
+
+/**
+ * Standalone repository class to access event streams. To be used in integration testing
+ */
+public class TestEventLogJdbcRepository extends EventJdbcRepository {
```
Comment (matt-rich) [inline stream-transformation-test/stream-transformation-it/src/test/java/uk/gov/justice/framework/tools/replay/TestEventLogJdbcRepository.java]: Do these classes not exist in the framework for us just to depend upon? or do we need to duplicate.

File: stream-transformation-test/stream-transformation-it/src/test/resources/liquibase/viewstore-db-changesets/001-event-log.table.changelog.xml [hunk 19]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+         http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.3.xsd">
+
+    <changeSet id="event-store-001" author="TechPod"
```
Comment (matt-rich) [inline stream-transformation-test/stream-transformation-it/src/test/resources/liquibase/viewstore-db-changesets/001-event-log.table.changelog.xml]: Do we need to duplicate these or can we just depend upon / maven copy the framework liquibase?

File: stream-transformation-tool-service/pom.xml [hunk 20]
```diff
@@ -31,6 +31,12 @@
             <groupId>uk.gov.justice.services</groupId>
             <artifactId>test-utils-core</artifactId>
             <scope>test</scope>
+            <exclusions>
```
Comment (matt-rich) [inline stream-transformation-tool-service/pom.xml]: as above, we'll need to investigate why we need to exclude this version. Also managed dependencies means we'd only need to set it once, hopefully.

File: stream-transformation-test/stream-transformation-it/src/test/java/uk/gov/justice/framework/tools/replay/StreamTransformationIT.java (hunk 13 above)
Comment (DesMarshall01) [inline stream-transformation-test/stream-transformation-it/src/test/java/uk/gov/justice/framework/tools/replay/StreamTransformationIT.java]: Would be better named createCommandToExecuteStreamTransformation()

File: stream-transformation-test/stream-transformation-it/src/test/resources/standalone-ds.xml [hunk 21]
```diff
@@ -0,0 +1,236 @@
...
+            </formatter>
+            <formatter name="COLOR-PATTERN">
+                <pattern-formatter pattern="%K{level}%d{HH:mm:ss,SSS} %-5p [%c] (%t) %s%e%n"/>
+            </formatter>
+        </subsystem>
+        <subsystem xmlns="urn:jboss:domain:bean-validation:1.0"/>
+        <subsystem xmlns="urn:jboss:domain:datasources:4.0">
+            <datasources>
+                <datasource jndi-name="java:/app/event-tool/DS.eventstore"
+                            pool-name="DS.replay-tool.eventstore" enabled="true"
```
Comment (DesMarshall01) [inline stream-transformation-test/stream-transformation-it/src/test/resources/standalone-ds.xml]: Change pool name to DS.event-tool.eventstore

File: stream-transformation-test/stream-transformation-it/src/test/resources/test.properties [hunk 22]
```diff
@@ -0,0 +1,10 @@
+db.eventstore.url=jdbc:h2:tcp://localhost:8092/mem:eventstore;MVCC=true
+db.viewstore.url=jdbc:h2:tcp://localhost:8092/mem:viewstore;MVCC=true
```
Comment (DesMarshall01) [inline stream-transformation-test/stream-transformation-it/src/test/resources/test.properties]: I don't think this is used

File: stream-transformation-test/stream-transformation-it/src/test/java/uk/gov/justice/framework/tools/replay/StreamTransformationIT.java [hunk 23]
```diff
@@ -179,11 +178,13 @@ private static DataSource initDatabase(final String dbUrlPropertyName,
         return dataSource;
     }
 
-    public boolean eventStoreTransformedEventPresent() throws SQLException {
+    private boolean eventStoreTransformedEventPresent() throws SQLException {
         final Stream<Event> eventLogs = EVENT_LOG_JDBC_REPOSITORY.findAll();
         final Optional<Event> event = eventLogs.filter(item -> item.getStreamId().equals(STREAM_ID)).findFirst();
-
-        return event.isPresent();
+        if (event.isPresent() && event.get().getName().equals("sample.events.transformedName")) {
```
Comment (matt-rich) [inline stream-transformation-test/stream-transformation-it/src/test/java/uk/gov/justice/framework/tools/replay/StreamTransformationIT.java]: you can just return the line, don't need to return true on the IF and false on the ELSE.

**Pull request merged**